docs/source/format/Intro.rst

```diff
@@ -0,0 +1,507 @@
+
+Also, the map stores the struct in a list and needs an offset as the list is
+variable shape.
+
+.. figure:: ./images/map-diagram.svg
+   :alt: Diagram is showing the difference between the map data type
+         presented in a Table and the data actually stored in computer
+         memory.
+
+   Physical layout diagram for map data type.
+
+Union
+-----
+
+The union is a nested data type where each slot in the union has a value with a data type
+chosen from a subset of possible Arrow data types. That means that a union array represents
+a mixed-type array. Unlike other data types, unions do not have their own validity bitmap
+and the nullness is determined by the child arrays.
+
+Arrow defines two distinct union data types, "dense" and "sparse".
+
+Dense Union
+^^^^^^^^^^^
+
+A Dense Union has one child array for each data type present in the mixed-type array and
+two buffers of its own:
+
+* **Types buffer:** holds data type id for each slot of the array. Data type id is
+  frequently the index of the child array; however, the relationship between data type
+  ID and the child index is a parameter of the data type.
+* **Offsets buffer:** holds relative offset into the respective child array for each
+  array slot.
+
+.. figure:: ./images/dense-union-diagram.svg
+   :alt: Diagram is showing the difference between the dense union data type
+         presented in a Table and the data actually stored in computer
+         memory.
+
+   Physical layout diagram for dense union data type.
+
+Sparse union
+^^^^^^^^^^^^
+
+A sparse union has the same structure as a dense union, with the omission of the offsets
+buffer. In this case, the child arrays are each equal in length to the length of the union.
+
+
+.. figure:: ./images/sparse-union-diagram.svg
+   :alt: Diagram is showing the difference between the sparse union data type
+         presented in a Table and the data actually stored in computer
+         memory.
+
+   Physical layout diagram for sparse union data type.
+
+Dictionary Encoded Layout
+=========================
+
+Dictionary encoding can be effective when one has data with many repeated values.
+The values are represented by integers referencing a dictionary usually consisting of
+unique values.
+
+.. figure:: ./images/dictionary-diagram.svg
+   :alt: Diagram is showing the difference between the dictionary data type
+         presented in a Table and the data actually stored in computer
+         memory.
+
+   Physical layout diagram for dictionary data type.
+
+Run-End Encoded Layout
+======================
+
+Run-end encoding is well-suited for representing data containing sequences of the
+same value. These sequences are called runs. Run-end encoded array has no buffers
+of its own, but has two child arrays:
+
+*  **Run ends array:** holds the index in the array where each run ends. The run ends
+   array always begins with 0 and contains one more element than the length of
+   its parent array.
```

Comment:
This is incorrect. Unlike `offsets`, run-end store the end of the run. And the number of run-ends is the same as length.